src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -344,15 +347,23 @@ public void onClose(int i, String s) {
                             "Connection was rejected, you should increase the Max connections to Kubernetes API", e);
                 }
 
+                boolean hasStarted = false;
                 try {
-                    started.await();
+                    // prevent a wait forever if the connection is closed as the listener would never be called
+                    hasStarted = started.await(WEBSOCKET_CONNECTION_TIMEOUT, TimeUnit.SECONDS);
                 } catch (InterruptedException e) {
                     closeWatch(watch);
                     throw new IOException(
                             "Interrupted while waiting for websocket connection, you should increase the Max connections to Kubernetes API",
                             e);
                 }
 
+                if (!hasStarted) {
+                    closeWatch(watch);
+                    throw new IOException(
+                            "Websocket connection was not established. This probably means the connection was closed already");
```

Comment:
Maybe mention the system property name to tune the timeout. I had reports of people running against a Kubernetes cloud with 10s ping (yes, crazy!) and this would be confusing if not mentioned.